### ISSUE TITLE
sort servers by latency

### ISSUE THREAD

**Issue by milahu** [lalanza808/monero.fail#34]
low latency is better

maybe this ranking could be implemented in browser javascript

<details>
<summary>
monero-remote-node-netselect.sh
</summary>

```sh
#!/usr/bin/env bash

# show the top 10 of monero remote nodes
# https://monero.fail/
# https://www.getmonero.org/resources/user-guides/remote_node_gui.html

list_url=https://monero.fail/haproxy.cfg
list_path=$HOME/.cache/monero.fail/haproxy.cfg
top_n=10
test_time=10

mkdir -p ${list_path%/*}

if [ -e "$list_path" ]; then
  ctime=$(stat -c%W "$list_path")
  now=$(date --utc +%s)
  age=$((now - ctime))
  if ((age > 10*24*60*60)); then
    echo "deleting old server list"
    rm "$list_path"
  fi
fi

if ! [ -e "$list_path" ]; then
  echo "fetching server list"
  curl -o "$list_path" "$list_url"
fi

echo "note: netselect requires root access"

cat "$list_path" |
grep "^    server " |
cut -d" " -f7 |
grep -v -E '\.(i2p|onion)(:[0-9]+)?$' |
sudo xargs netselect -t$test_time -s$top_n
```

</details>

**Comment by lalanza808** [comment]
latency from the server or from the user?

**Comment by milahu** [comment]
from the user

**Comment by plowsof** [comment]
or location flag? i think you want 'fastest servers'? maybe you ping 20ms to one but its running on terrible hardware and slower than a node which has 200 ping. this would require client side scripting (e.g. js) to ping -all- servers when you visit the page, i think alot of users wouldnt be keen on that

**Comment by milahu** [comment]
> when you visit the page

no, the user would have to manually start such a `netselect.js` process

see also

- https://github.com/alfg/ping.js
- https://github.com/jdfreder/pingjs
- [How to ping IP addresses using JavaScript](https://stackoverflow.com/questions/4954741/how-to-ping-ip-addresses-using-javascript)
- [Is it possible to ping a server from the browser?](https://stackoverflow.com/questions/4282151/is-it-possible-to-ping-a-server-from-the-browser)
- [How to ping a server using JavaScript ?](https://www.geeksforgeeks.org/javascript/how-to-ping-a-server-using-javascript/)

**Comment by lalanza808** [comment]
okay, check it out https://monero.fail/find

!40

**Comment by milahu** [comment]
yay! : )

> Nodes that cannot be pinged are listed at the bottom.

why does ping fail on some nodes?
(in my case, munich germany, ping works for 1 of 20 nodes)
(the webui should explain in a footnote/modal why ping fails on some nodes)

also, when i click the map (to set my location) too often
then the "Pinging nodes" progress animation can hang like `"Pinging nodes... 3 / 4"`
showing [sandwich.gif](https://monero.fail/static/images/sandwich.gif) forever
(in very rare cases)

**Comment by lalanza808** [comment]
only servers that support CORS can be "pinged" by the browser for security reasons. 

<img width="747" height="84" alt="Image" src="https://github.com/user-attachments/assets/e7b0508f-bac3-4f2e-9a2a-d2e5d7693c42" />

**Comment by lalanza808** [comment]
i feel like this is good enough for now. if you have any other ideas for improvements please make a new issue. tyvm for the support
